### PR TITLE
[Feature] Namespace store filesystem update

### PR DESCRIPTION
**Pull request by vbnrh**
Design: https://marvelapp.com/prototype/6884a1f/screen/81852271

Epic: https://issues.redhat.com/browse/RHSTOR-1953

Screenshot from 2021-11-29 13-30-35

Signed-off-by: Vineet Badrinath <vineetbnath@gmail.com>

### CODE REVIEW

**Comment by SanjalKatiyar** [comment]
For OCP 4.10 + ODF 4.9, do we still want to show this `Filesystem` dropdown item ?
If no, then we might need a feature guard.

File: frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx [hunk 1]
```diff
@@ -176,6 +184,45 @@ const NamespaceStoreForm: React.FC<NamespaceStoreFormProps> = withHandlePromise<
...
+        <>
+          <FormGroup
+            label={t('ceph-storage-plugin~Persistent volume claim')}
+            fieldId="pvc-name"
+            className="nb-endpoints-form-entry"
+            isRequired
+          >
+            <PVCDropdown
+              dataTest="pvc-dropdown"
+              namespace={namespace}
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx:198]:
```suggestion
              id="pvc-name"
              namespace={namespace}
```

**Comment by vbnrh** [comment]
> For OCP 4.10 + ODF 4.9, do we still want to show this `Filesystem` dropdown item ? If no, then we might need a feature guard.

No need of feature gating . This feature is in the backend since 5.8

File: frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx (hunk 1 above)
Comment (vbnrh) [reply frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx:199]: are you asking to remove the namespace ?

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx:199]: nope... to add `id` corresponding to `fieldId` added in the `FormGroup`

Comment (vbnrh) [reply frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx:199]: It is there ? :O

**Review by SanjalKatiyar — APPROVED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx [hunk 2]
```diff
@@ -83,7 +89,7 @@ const NamespaceStoreForm: React.FC<NamespaceStoreFormProps> = withHandlePromise<
         ssl: false,
       },
     };
-    if (externalProviders.includes(provider)) {
+    if (provider !== BC_PROVIDERS.FILESYSTEM && externalProviders.includes(provider)) {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx:92]: Why is this required? Is `FileSystem` an external provider? Could we perhaps update `getExternalProviders`?

Comment (vbnrh) [reply frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx:92]: I don't want those specs for Filesystem type. No buckets/secrets for filesystem type as they use native PVCs

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx:92]: then remove `FileSystem` from `externalProviders`. I don't see the need to have two conditions when this logic can be added somewhere else.

Comment (vbnrh) [reply frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx]: removed

**Review by bipuladh — APPROVED** [review]
/lgtm
/label px-approved
/label doc-approved

**Comment by SanjalKatiyar** [comment]
tested !!
UI is abiding by the proposed UXD designs and PVC dropdown is showing RWX PVCs only.
Screenshot from 2021-12-13 16-38-59

**Comment by SanjalKatiyar** [comment]
Based on above comment.

/label qe-approved
